modules/geo/src/main/java/org/opensearch/geo/search/aggregations/bucket/geogrid/util/GeoShapeHashUtil.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+package org.opensearch.geo.search.aggregations.bucket.geogrid.util;
+
+import org.opensearch.common.geo.GeoShapeDocValue;
+import org.opensearch.geometry.Rectangle;
+import org.opensearch.geometry.utils.Geohash;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * We have a {@link Geohash} class present at the libs level, not using that because while encoding the shapes we need
+ * {@link GeoShapeDocValue}. This class provided the utilities encode the shape as GeoHashes
+ */
+public class GeoShapeHashUtil {
+
+    /**
+     * The function encodes the shape provided as {@link GeoShapeDocValue} to a {@link List} of {@link Long} values
+     * (representing the GeoHashes) which are intersecting with the shapes at a given precision.
+     *
+     * @param geoShapeDocValue {@link GeoShapeDocValue}
+     * @param precision int
+     * @return {@link List} containing encoded {@link Long} values
+     */
+    public static List<Long> encodeShape(final GeoShapeDocValue geoShapeDocValue, final int precision) {
+        final List<Long> encodedValues = new ArrayList<>();
+        final GeoShapeDocValue.BoundingRectangle boundingRectangle = geoShapeDocValue.getBoundingRectangle();
+        long topLeftGeoHash = Geohash.longEncode(boundingRectangle.getMinX(), boundingRectangle.getMaxY(), precision);
+        long topRightGeoHash = Geohash.longEncode(boundingRectangle.getMaxX(), boundingRectangle.getMaxY(), precision);
+        long bottomRightGeoHash = Geohash.longEncode(boundingRectangle.getMaxX(), boundingRectangle.getMinY(), precision);
+
+        // Now iterate these 4 coordinates.
```

Comment:
Out of curiosity, I would have expected to see `bottomLeftGeoHash` somewhere but there are only 3 coordinates, not 4, typo?

Reply:
Actually ```bottomLeftGeoHash``` variable is not required. So the way this logic works is we are starting from topLeft and for every row we move till right most cell. and the loop is breaking when we reach the bottomRight cell.

Seems like the comment is misleading. I am using 3 coordinates but saying iterate over 4. Let me change it.

Reply:
I have removed that comment.